TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -16,10 +18,120 @@ public static class SavepointStatementChecker
                 if (savePointName.StartsWith("SYS", StringComparison.OrdinalIgnoreCase))
                 {
                     DiagnosticUtils.AddError(savepointStatement,
-                        "Invalid savepoint-name, it must not begin with 'SYS'.",
+                        "Invalid savepoint-name, it must not begin with 'SYS'.", context);
+                }
+            }
+        }
+    }
+
+    internal static class AlterSequenceStatementChecker
+    {
+        public static void OnCodeElement(AlterSequenceStatement alterSequenceStatement,
+            List<string> duplicatedClauses, bool emptyClauseSet,
+            CodeElementsParser.AlterSequenceStatementContext context)
+        {
+            foreach (var clauseType in duplicatedClauses)
+            {
+                DiagnosticUtils.AddError(alterSequenceStatement,
+                    clauseType + " cannot be specified more than once",
+                    context);
+            }
+
+            if (emptyClauseSet)
+            {
+                DiagnosticUtils.AddError(alterSequenceStatement,
+                    "At least one option among RESTART, INCREMENT, MINVALUE, MAXVALUE, CYCLE, CACHE or ORDER must be specified",
+                    context);
+            }
+
+            if (alterSequenceStatement.MinValue != null)
```

Comment:
You have to factorize code.

```csharp
void CheckIsInteger(string optionName, Token optionValue)
{
    [...]
    - retrieve literal value
    - check type
    - check digits if decimal
}
```

you can also return the value as `double` if you have further checks to do.

instead of copy pasting